File: tests/test_cases/kernel/cuda/test_cuda.cpp
/*
 * @Author       : Thinksky5124
 * @Date         : 2024-04-07 13:57:12
 * @LastEditors  : Thinksky5124
 * @LastEditTime : 2024-04-20 16:37:31
 * @Description  : file content
 * @FilePath     : /AIHPC-Larning/tests/test_cases/kernel/cuda/test_cuda.cpp
 */
#include "gtest/gtest.h"
#include "../../../aihpc/include/aihpc.h"

TEST(TestAddInt, test_add_int_1) {
  kernel::cuda::vector_add();
  EXPECT_EQ(1, 1);
}

File: tests/test_cases/kernel/cuda/test_cuda.py
'''
Author       : Thinksky5124
Date         : 2024-04-19 10:42:16
LastEditors  : Thinksky5124
LastEditTime : 2024-04-21 20:52:17
Description  : file content
FilePath     : /AIHPC-Larning/tests/test_cases/kernel/cuda/test_cuda.py
'''
import aihpc
import torch

class TestCUDA:
    def test_add(self):
        a = torch.randn(3, 3)
        b = torch.randn(3, 3)
        c_c = a + b
        c = aihpc._C.kernel.cuda.add(a, b, False)
        assert torch.allclose(c, c_c)